### ISSUE TITLE
Can't open Storage Explorer after reloading it with 'Use app proxy settings' is configured

### ISSUE THREAD

**Issue by v-xianya** [microsoft/AzureStorageExplorer#4705]
**Storage Explorer Version**: 1.21.0-dev
**Build Number**: 20210723.3
**Branch**: main
**Platform/OS**: Windows 10
**Architecture**: ia32
**How Found**: Exploratory testing
**Regression From**: Previous release(1.20.1)

## Steps to Reproduce ##
1. Launch Storage Explorer.
2. Download Fiddler from https://telerik.com/download/fiddler.
3. Configure Fiddler from [Step 3-16] 'https://docs.microsoft.com/en-us/azure/storage/common/storage-explorer-troubleshooting?tabs=Windows%2C2004#network-logs'.
4. Go to 'Settings' -> Select "Use app proxy settings", and set the URL to http://localhost and the port to 8888.
5. Reload Storage Explorer.
6. Check whether succeed to reload.

## Expected Experience ##
Succeed to reload.

## Actual Experience ##
Storage Explorer can't be opened.

## Additional Context ##
1. There is only one cert in the Path 'C:\Users\usename\AppData\Roaming\StorageExplorer\certs'.
2. Try to open Storage Explorer from cmd. Show the `'Warning: Accessing non-existent property 'ErrorKind' of module exports inside circular dependency
(Use `StorageExplorer --trace-warnings ...` to show where the warning was created)'.`

![image](https://user-images.githubusercontent.com/41351993/126752225-ee3e9bd4-1fe4-4031-9fc1-c89f36257b83.png)

**Comment by craxal** [comment]
@v-xianya Can you do what the warning says and run `StorageExplorer --trace-warnings` to see what this prints out?

**Comment by v-xianya** [comment]
Hi @craxal I just found an easier way to repro this issue😥 (build #20210727.2). Below are the steps:
1. Launch Storage Explorer.
2. Open 'Settings -> Application -> Proxy' -> Select 'Use app proxy settings'. (Keep the default Hostname empty and Port: 80)
3. Close Storage Explorer -> Try to open Storage Explorer.

**Actual:** Storage Explorer can't be opened.

Then try to open Storage Explorer from cmd by running command 'StorageExplorer.exe'. 
![image](https://user-images.githubusercontent.com/41351993/127102731-f25f7f02-3faa-4e36-b1d1-defaaaac16bf.png)

Run command 'StorageExplorer --trace-warnings ...'. the results as below. It is the same as the above screenshot.
![image](https://user-images.githubusercontent.com/41351993/127103060-76d1fe8b-a1dc-4d0a-8c64-71b4b87bad20.png)
Here is content in the 2021-07-26_225424_main.log:
`[2021-07-27T05:54:24.100Z] ( main) <INFO> Startup app state:  {
  lastWindowState: { rect: { x: 0, y: 0, width: 1000, height: 768 }, maximized: true },
  disableGpu: false,
  dialogAutoResponse: {}
}`

**Comment by craxal** [comment]
@v-xianya Um...hostname shouldn't be empty. What happens if you open the settings file and enter something for the hostname (doesn't have to be a real proxy)?
```json
"application.proxy.hostname": "a"
```

**Comment by v-xianya** [comment]
Hi @craxal we added  "application.proxy.hostname": "a" to settings file. This issue still reproduces.

In addition, we also tried the below steps.  this issue still reproduces.
1. Without download Fiddler
1. Launch Storage Explorer with 'Do not use proxy'.
2. Switch to 'Use app proxy setting' [Protocol: https, Input "a" to Hostname, Port: 80]
3. Closed Storage Explorer.
4. Then try to start Storage Explore -> Can't start Storage Explorer.
